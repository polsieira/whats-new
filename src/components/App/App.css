* {
  box-sizing: border-box;
  margin: 0;
  padding: 0;
}

.App {
  display: flex;
  background: #141527;
  text-align: center;
  width: 100%;
}
