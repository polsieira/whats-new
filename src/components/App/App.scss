* {
  box-sizing: border-box;
  margin: 0;
  padding: 0;
}

.App {
  display: flex;
  background: #141527;
  padding: 20px;
  text-align: center;
  width: 100%;
}

.wrapper--news{
  width: 80%;
}
