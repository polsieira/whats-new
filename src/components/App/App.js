import React, { Component } from 'react';
import local from '../../data/local';
import './App.css';
import Menu from '../Menu/Menu';
import SearchForm from '../SearchForm/SearchForm';
import NewsContainer from '../NewsContainer/NewsContainer';

class App extends Component {
  constructor() {
    super();
    this.state = {
      local
    }
  }

  render() {
    return (
      <div className="App">
        <Menu />
        <SearchForm />
        <NewsContainer local={this.state.local} />
      </div>
    );
  }
}

export default App;
