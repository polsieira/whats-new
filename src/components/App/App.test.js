import React from 'react';
import { shallow } from "enzyme";
import ReactDOM from 'react-dom';
import App from './App';
import health from '../../data/health';
import local from '../../data/local';

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<App />, div);
  ReactDOM.unmountComponentAtNode(div);
});

describe('App', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(<App />)
  });

  it('should match the snapshot with all data passed in correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should update state current category and displayed articles when change state is called', () => {
    //Setup
    const mockEvent = {
      target: {
        id: 'health'
      }
    };
    const expected = health;

    //Execution
    wrapper.instance().changeState(mockEvent);

    //Expectation
    expect(wrapper.state('currentCategory')).toEqual(expected);
  });

  it('should search articles and only display results', () => {
    //Setup
    const mockSearchedText = 'Chicken';

    wrapper.setState({
      currentCategory: local,
      displayedArticles: local
    });

    //Execution
    wrapper.instance().searchArticles(mockSearchedText);

    //Expectation
    expect(wrapper.state('displayedArticles')).toEqual()
  });
});
