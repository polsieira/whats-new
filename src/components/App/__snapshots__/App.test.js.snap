// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should match the snapshot with all data passed in correctly 1`] = `
<div
  className="App"
>
  <Menu
    changeState={[Function]}
  />
  <div
    className="wrapper--news"
  >
    <SearchForm
      searchArticles={[Function]}
    />
    <NewsContainer
      displayedArticles={Array []}
    />
  </div>
</div>
`;
